mkosi/distributions/gentoo.py

```diff
@@ -211,125 +118,142 @@ def fetch_fix_stage3(self) -> None:
+        if not cls.stage3_cache.joinpath(".cache_isclean").exists():
+            with complete_step(f"Extracting {stage3_tar.name} to {cls.stage3_cache}"):
                 run([
                     "tar",
                     "--numeric-owner",
-                    "-C", stage3_tmp_extract,
+                    "-C", cls.stage3_cache,
                     "--extract",
                     "--file", stage3_tar_path,
                     "--exclude", "./dev",
+                    "--exclude", "./proc",
                 ])
 
-                unlink_try_hard(stage3_tmp_extract.joinpath("dev"))
-                unlink_try_hard(stage3_tmp_extract.joinpath("proc"))
-                unlink_try_hard(stage3_tmp_extract.joinpath("sys"))
-
-                stage3_tmp_extract.joinpath(".cache_isclean").touch()
-
-        log_step(f"Copying {stage3_tmp_extract} to {self.root}")
-        copy_path(stage3_tmp_extract, self.root)
-
-    def set_useflags(self) -> None:
-        package_use = self.portage_cfg_dir / "package.use"
-        package_use.mkdir(exist_ok=True)
-
-        package_use.joinpath("systemd").write_text(
-            # repart for usronly
-            dedent(
-                """\
-                # MKOSI: used during the image creation
-                # "/usr/lib/systemd/boot/efi": No such file or directory
-                sys-apps/systemd gnuefi
-                sys-apps/systemd -cgroup-hybrid
-                sys-apps/systemd elfutils # for coredump
-
-                sys-apps/systemd homed cryptsetup -pkcs11
-                # See: https://bugs.gentoo.org/832167
-                sys-auth/pambase homed
-
-                # MKOSI: usronly
-                sys-apps/systemd repart
-                # MKOSI: make sure we're init (no openrc)
-                sys-apps/systemd sysv-utils
-                """
+            for d in ("binpkgs", "distfiles", "repos"):
+                state.cache_dir.joinpath(d).mkdir(exist_ok=True)
+
+            config = state.pkgmngr / "etc/portage"
+
+            package_use = config / "package.use"
+            package_use.mkdir(parents=True, exist_ok=True)
+            package_use.joinpath("systemd").write_text(
+                # repart for usronly
+                dedent(
+                    """\
+                    # MKOSI: used during the image creation
+                    # "/usr/lib/systemd/boot/efi": No such file or directory
+                    sys-apps/systemd gnuefi
+                    sys-apps/systemd -cgroup-hybrid
+                    sys-apps/systemd elfutils # for coredump
+
+                    sys-apps/systemd homed cryptsetup -pkcs11
+                    # See: https://bugs.gentoo.org/832167
+                    sys-auth/pambase homed
+
+                    # MKOSI: usronly
+                    sys-apps/systemd repart
+                    # MKOSI: make sure we're init (no openrc)
+                    sys-apps/systemd sysv-utils
+                    """
+                )
```

Comment:
I don't think we should hardcode this into mkosi. systemd might not even be installed into the image, so we shouldn't be configuring it. This should be configured by the distribution.

Reply:
`gnuefi` is a must or else `mkosi` fails.
other stuff are from before `usr-merge` era. let make an issue out of this so i remember to come back to it once we have landed this.

Reply:
https://github.com/systemd/mkosi/issues/1640